tdrs-backend/tdpservice/search_indexes/parsers/tanf_validators.py

```diff
@@ -0,0 +1,191 @@
+# https://www.acf.hhs.gov/sites/default/files/documents/ofa/tanf_warning_edits_section_1.pdf
+
+def t1_097(model_obj):
+    """Validate funding stream."""
+    return model_obj.FUNDING_STREAM in [1, 2]
+
+def t1_099(model_obj):
+    """Validate receives subsidized housing."""
+    return model_obj.RECEIVES_SUB_HOUSING in [1, 2, 3]
+
+def t1_100(model_obj):
+    """Validate receives medical assistance."""
+    return model_obj.RECEIVES_MED_ASSISTANCE in [1, 2]
+
+def t1_101(model_obj):
+    """Validate receives food stamps."""
+    return model_obj.RECEIVES_FOOD_STAMPS in [1, 2]
+
+def t1_102(model_obj):
+    """Validate amount of food stamp assistance."""
+    return model_obj.AMT_FOOD_STAMP_ASSISTANCE >= 0
+
+def t1_103(model_obj):
+    """Validate amount of subsidized child care."""
+    return model_obj.AMT_SUB_CC >= 0
+
+def t1_104(model_obj):
+    """Validate amount of child support."""
+    return model_obj.CHILD_SUPPORT_AMT >= 0
+
+def t1_105(model_obj):
+    """Validate amount of family's cash resources."""
+    return model_obj.FAMILY_CASH_RESOURCES >= 0
+
+def t1_107(model_obj):
+    """Validate cash and cash equivalents."""
+    return model_obj.CASH_AMOUNT >= 0 and model_obj.NBR_MONTHS >= 0
+
+def t1_108(model_obj):
+    """Validate tanf child care."""
+    return model_obj.CC_AMOUNT >= 0 and model_obj.CHILDREN_COVERED >= 0
+
+def t1_110(model_obj):
+    """Validate transportation."""
+    return model_obj.TRANSP_AMOUNT >= 0 and model_obj.TRANSP_NBR_MONTHS >= 0
+
+def t1_112(model_obj):
+    """Validate transitional services."""
+    return model_obj.TRANSITION_SERVICES_AMOUNT >= 0 and model_obj.TRANSITION_NBR_MONTHS >= 0
+
+def t1_114(model_obj):
+    """Validate other."""
+    return model_obj.OTHER_AMOUNT >= 0 and model_obj.OTHER_NBR_MONTHS >= 0
+
+def t1_117(model_obj):
+    """Validate reason for & amount of assistance reductions."""
+    return model_obj.RECOUPMENT_PRIOR_OVRPMT >= 0
+
+def t1_121(model_obj):
+    """Validate waiver evaluation experimental & control groups."""
+    return model_obj.WAIVER_EVAL_CONTROL_GRPS in [9, '']
+
+def t1_122(model_obj):
+    """Validate is tanf family exempt from federal time limit provisions."""
+    return model_obj.FAMILY_EXEMPT_TIME_LIMITS in [1, 2, 3, 4, 6, 7, 8, 9]
+
+def t1_123(model_obj):
+    """Validate is tanf family a new child only family."""
+    return model_obj.FAMILY_NEW_CHILD in [1, 2]
+
+# T1 Category 3 TANF Fatal Edits
+# https://www.acf.hhs.gov/sites/default/files/documents/ofa/tanf_fatal_edits_sections_1_and_4.pdf
+
+def t1_004(model_obj):
+    """Validate case number."""
+    return model_obj.CASE_NUMBER != ''  # TODO not sure how to check a blank char, Is this cat 1?
+
+def t1_009(model_obj):
+    """Validate disposition.
+
+    TODO add check for item 1
+    TODO make sure blank char is valid
+    """
+    if model_obj.DISPOSITION == 2:
+        return model_obj.CASE_NUMBER != '' and model_obj.STRATUM != '' and model_obj.RPT_MONTH_YEAR != ''
```

Comment:
I am not sure what a black field will come in as so this needs to be double checked.

Reply:
@ADPennington We should sync on testing a document where this error is relevant.

Reply:
Sure! Was just chatting with @reitermb about this. we mostly want to ensure that this field is 11 alphanumeric characters that aren't all spaces or all underscores.